spaceallocator/spaceallocator.go

```diff
@@ -374,11 +374,10 @@ func (allocator *allocatorInstance) removeOutdatedItems(requiredSize uint64) (fr
 
 	sort.Sort(byTimestamp(allocator.part.outdatedItems))
 
-	i := 0
+	for i := 0; i < len(allocator.part.outdatedItems); {
+		item := &allocator.part.outdatedItems[i]
 
-	for _, item := range allocator.part.outdatedItems {
-		if item.allocator != allocator {
-			allocator.part.outdatedItems[i] = item
+		if allocator.part.outdatedItems[i].allocator != allocator {
```

Comment:
What was the main issue? It is ok to remove items from other allocators if they are outdated.